tests/moe/test_trellis_checkpoint_bridge.py

```
@@ -0,0 +1,344 @@
+    counts = []
+    for tier in range(len(assembled.tier_bits)):
+        gate_ids = [e for e in range(experts) if assembled.gate_tiers[e] == tier]
+        up_ids = [e for e in range(experts) if assembled.up_tiers[e] == tier]
+        down_ids = [e for e in range(experts) if assembled.down_tiers[e] == tier]
+        slots = max(len(gate_ids), len(up_ids), len(down_ids))
+        counts.append(slots)
+        bits = assembled.tier_bits[tier]
+
+        def _padded(stack_map, fc1: bool) -> torch.Tensor:
+            shape = (
+                (_HIDDEN // 16, local // 16, 16 * bits)
```

Ruff (0.16.2)

[warning] Function definition does not bind loop variable `bits`

(B023)